kolibri_explore_plugin/views.py

```diff
@@ -194,25 +240,36 @@ def get(self, request):
         )
 
     def post(self, request):
+        grade = "primary"
         collection = "small"
         if request.body:
             data = json.loads(request.body)
             collection = data.get("collection", "small")
+            grade = data.get("grade", "primary")
 
-        token = self.COLLECTIONS[collection]["token"]
-
-        channel_viewset = RemoteChannelViewSet()
-        channels = channel_viewset._make_channel_endpoint_request(
-            identifier=token
+        collection_manifest = os.path.join(
+            COLLECTION_PATHS, f"{grade}-{collection}.json"
         )
 
-        job_ids = []
+        if os.path.exists(collection_manifest):
+            manifest = {}
+            with open(collection_manifest) as f:
+                manifest = json.load(f)
+            channels = manifest.get("channels", [])
+        else:
+            # Fallback, download full collection using the token
+            token = self.COLLECTIONS[grade][collection]["token"]
+
+            channel_viewset = RemoteChannelViewSet()
+            channels = channel_viewset._make_channel_endpoint_request(
+                identifier=token
+            )
```

Comment:
Yes, it's better to remove this fallback, because as you said it could hide some error and will make it more difficult to debug. I'll replace with an error in the interface.